genie-agent/src/main/java/com/netflix/genie/agent/execution/ExecutionAutoConfiguration.java

```diff
@@ -32,14 +86,186 @@
 public class ExecutionAutoConfiguration {
 
     /**
-     * Provide a lazy execution context bean if one hasn't already been defined.
+     * Bean identifier for state machine used to run a job.
+     */
+    public static final String STATE_MACHINE_BEAN_NAME = "stateMachine";
+
+    /**
+     * Bean identifier for job execution stages.
+     */
+    public static final String EXECUTION_STAGES_BEAN_NAME = "jobExecutionStages";
+
+    /**
+     * Provide a lazy {@link LoggingListener} bean.
+     *
+     * @return A {@link LoggingListener} instance
+     */
+    @Bean
+    @Lazy
```

Comment:
I did a quick test with this vs. global lazy agent initialization. 
The latter is always 5-10% slower running a NOOP job.
And 5-10% faster running a non-job command (i.e. `help`).

Without no kind of lazy initialization, it's actually faster in both cases, so gonna go with that since it's also the "safer" alternative.

Reply:
Turns out, the lazy initialization (of gRPC components specifically) is required for things to bind properly (for example gRPC channel to use the hostname and port specified in the CLI args).

So removing lazy won't work, i would need to make those beans into factories.
For the moment, gonna leave this alone.